src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -16,7 +16,12 @@
 
 namespace Microsoft.CodeAnalysis.FindSymbols
 {
-    using RelatedTypeCache = ConditionalWeakTable<Solution, ConcurrentDictionary<(SymbolKey, IImmutableSet<Project>), AsyncLazy<ImmutableArray<(SymbolKey, ProjectId)>>>>;
+    // Must cache using SymbolKey+ProjectId.  That's because the same symbol key may be found among many projects, but
+    // the same operation on the same symbol key might produce different results depending on which project it was found
+    // in.  For example, each symbol's project may have a different set of downstream dependent projects.  As such,
+    // there may be a different set of related symbols found for each.
+    using RelatedTypeCache = ConditionalWeakTable<Solution, ConcurrentDictionary<(SymbolKey, ProjectId, IImmutableSet<Project>), AsyncLazy<ImmutableArray<(SymbolKey, ProjectId)>>>>;
```

Comment:
❔ In what case would you have two keys with the same `SymbolKey` and `IImmutableSet<Project>`, but a different `ProjectId` and mapped value?

Reply:
so the `ImmutableSet<Project>` is what hte caller is passing in saying "only include results from these projects" (or, in other words, don't bother searching other projects).  

That's independent of the ProjectId/Symbol that was passed in.  For example, imagine:

Project A, with Symbol S.

Two people can use that exact same symbol searching for dependent symbols, but asking to search different project-subsets of the solution.